CHANGELOG.md

```diff
@@ -3,6 +3,7 @@
 ### Breaking Changes
 * Removed all references and API's releated to permissions. These are now managed through MongoDB Realm. Read more [here](XXX).
 * Removed Query Based Sync API's and Subscriptions. These API's are not initially supported by MongoDB Realm. They will be re-introduced in a future release. `SyncConfiguration.partionKey()` has been added as a replacement. Read more [here](XXX).  
+* Destructive updates of a schema of a synced Realm will now consistently throw an `UnsupportedOperationException` instead of some methods throwing `IllegalArgumentException`. The affected methods are `RealmSchema/MutableRealmSchema.{remove,rename}` and `RealmObjectSchema/MutableRealmObjectSchema.{setClassName,removeField,renameField,removeIndex,removePrimaryKey,addPrimaryKey,addField}` 
```

Comment:
3 things:

1) This doesn't sound right. `MutableRealmSchema` is used by local Dynamic Realms, Local typed and Synced Realms (both Dynamic and normal) use `ImmutableRealmSchema`.
2) `MutableRealmSchema/ImmutableRealmSchema` is hidden from the public API's. Users do now "know" of these classes. They only know of `RealmSchema` and `RealmObjectSchema`
3) Using `/` as seperator between class and method looks a bit weird. Normally we have used `Class.Method()` in the past`

Reply:
I tried to clean up. Remove the immutable/mutable classes and named all the methods explicitly.